File: RowndPluginEventEmitter.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(RowndPluginEventEmitter, RCTEventEmitter)
  RCT_EXTERN_METHOD(supportedEvents)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

@end

File: ios/RowndPluginEventEmitter.swift
import Rownd
import SwiftUI
import React

@objc(RowndPluginEventEmitter)
class RowndPluginEventEmitter: RCTEventEmitter {
    
    public static var emitter: RCTEventEmitter!

    override init() {
        super.init()
        RowndPluginEventEmitter.emitter = self
    }

    open override func supportedEvents() -> [String] {
      ["update_state","onReady", "onPending", "onFailure"]      // etc.
    }
  }

